content/faq/errors.md

```diff
@@ -0,0 +1,38 @@
+### Common errors
+
+During your development with NestJS, you may encounter various errors as you learn the framework.
+
+#### "Cannot resolve dependency" error
+
+Probably the most common error message is about Nest not being able to resolve dependencies of a provider. The error message usually looks something like this:
+
+```bash
+Nest can't resolve dependencies of the <provider> (?). Please make sure that the argument <unknown_token> at index [<index>] is available in the <module> context.
+
+Potential solutions:
+- If <unknown_token> is a provider, is it part of the current <module>?
+- If <unknown_token> is exported from a separate @Module, is that module imported within <module>?
+  @Module({
+    imports: [ /* the Module containing <unknown_token> */ ]
+  })
+```
+
+There are a few gotchas, that are common. One is putting a provider in an `imports` array. If this is the case, the error will have the provider's name where `<module>` should be.
```

Comment:
@jmcdo29 can we add a little bit more info here? I suppose that some people may visit this page without having *any* idea on how to fix this issue so perhaps we should mention obvious solutions as well

Reply:
I've added in a small blurb, hopefully it makes enough sense. It's a bit difficult to mention this in a generic way. Let me know if you think it needs any more improvement

Reply:
Thanks @jmcdo29, it made things a little bit more cleaner. Still, I believe that we should also add a very obvious explanation (at the beginning, right after the error message) - the most common one - which simply informs people that in 99% of cases they just forgot to register a provider in a module. We could link to this section https://docs.nestjs.com/fundamentals/custom-providers#di-fundamentals for more details.

Reply:
Ah, of course. Makes sense. Sometimes the obvious is hard to see once you get so into the weeds 😆 I'll get that added this evening and add the link to earlier in the docs. Thanks for the suggestion.